### PR TITLE
add kube-linter checks

### PR DESCRIPTION
**Pull request by sadlerap**
Infra-deployments has a suite of checks that makes sure every kustomize manifest committed passes kube-linter in its default configuration.  If we want to release onto platforms supported by infra-deployments, we need to be able to pass these checks too.

This introduces a new check in CI to ensure that we don't fall out-of-line with what infra-deployments expects.

### CODE REVIEW

File: operator/config/manager/kustomization.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: quay.io/konflux-workspaces/workspaces-operator
+  newTag: v0.1.0-alpha2
```
Comment (filariow) [inline operator/config/manager/kustomization.yaml:6]: should we revert this?

File: server/config/server/kustomization.yaml [hunk 2]
```diff
@@ -15,3 +17,7 @@ configMapGenerator:
   name: traefik-sidecar-dynamic-config
   options:
     disableNameSuffixHash: true
+images:
+- name: workspaces/rest-api
+  newName: quay.io/konflux-workspaces/workspaces-server
+  newTag: v0.1.0-alpha2
```
Comment (filariow) [inline server/config/server/kustomization.yaml:23]: ditto

File: server/config/default/kustomization.yaml [hunk 3]
```diff
@@ -24,15 +23,15 @@ replacements:
     targets:
       # create Role and RoleBinding to read SpaceBinding into toolchain-host-operator
       - options:
-          create: false
+          create: true
```
Comment (filariow) [inline server/config/default/kustomization.yaml:26]: Out of curiosity - why is this change?

**Review by filariow — CHANGES_REQUESTED** [review]
nice, just some nits

File: operator/config/manager/kustomization.yaml (hunk 1 above)
Comment (sadlerap) [reply operator/config/manager/kustomization.yaml:6]: Nope, we need it this time.  kube-linter denies any image tagged with `:latest`.

File: server/config/default/kustomization.yaml (hunk 3 above)
Comment (sadlerap) [reply server/config/default/kustomization.yaml]: Reverted in 8aeec18.

Comment (sadlerap) [reply server/config/default/kustomization.yaml]: It's needed since kustomize fails: https://github.com/konflux-workspaces/workspaces/actions/runs/9663771194/job/26656780312?pr=214

Comment (filariow) [reply server/config/default/kustomization.yaml:26]: ok

File: server/config/server/kustomization.yaml (hunk 2 above)
Comment (sadlerap) [reply server/config/server/kustomization.yaml]: Fixed in 21d3b5e.

File: operator/config/manager/kustomization.yaml (hunk 1 above)
Comment (sadlerap) [reply operator/config/manager/kustomization.yaml]: Fixed in 21d3b5e.

File: .github/workflows/security.yaml [hunk 4]
```diff
@@ -90,3 +90,42 @@ jobs:
...
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v4
+
+      - name: Render kustomize templates
+        run: |
+          mkdir out
+          find operator server -name 'kustomization.yaml' | \
+            xargs -I {} -n1 -P8 \
+            bash -c 'dir=$(dirname "{}"); output_file=$(echo $dir | tr / -)-kustomization.yaml; if ! log=$(kustomize build "$dir" -o "out/$output_file" 2>&1); then echo "Error when running kustomize build for $dir: $log" && exit 1;fi'
```
Comment (filariow) [inline .github/workflows/security.yaml:111]: what you think about overwriting here the image tag and revert the change in default kustomization files? I'd prefer to keep kustomization files as stock as possible.

```suggestion
            bash -c 'dir=$(dirname "{}"); output_file=$(echo $dir | tr / -)-kustomization.yaml; if ! log=$(cd "$dir" && kustomize edit set image workspaces/rest-api:index workspaces/controller:index && kustomize build . -o "out/$output_file" 2>&1); then echo "Error when running kustomize build for $dir: $log" && exit 1;fi'
```

File: .github/workflows/push.yaml
Comment (filariow) [inline .github/workflows/push.yaml:1]: Not really related to this PR, I'd propose to move the `Create draft release` job to a dedicated action

File: .github/workflows/security.yaml (hunk 4 above)
Comment (filariow) [reply .github/workflows/security.yaml:111]: or else we may use `make -C operator  package` and `make -C server package` to build the manifests and lint those

File: .github/workflows/push.yaml
Comment (sadlerap) [reply .github/workflows/push.yaml:1]: Yep, that's some tech debt I've been meaning to do, though I've not really wanted to copy-paste the workflows for building/pushing images.  I know there's a method for reusing workflows, but I've been wanting to get something working (even if it's suboptimal) before doing some cleanup.

File: .github/workflows/security.yaml (hunk 4 above)
Comment (sadlerap) [reply .github/workflows/security.yaml:111]: I pulled this line straight from infra-deployments, and I'd like to keep this in line with them as much as possible.  That said, this _should_ work.

Comment (sadlerap) [reply .github/workflows/security.yaml:111]: I'll also revert things to be back to `controller:latest` and `rest-api:latest` in the deployments.

Comment (sadlerap) [reply .github/workflows/security.yaml]: Fixed in d861b91.

**Review by filariow — APPROVED** [review]
